src/metabase/util.cljc

```diff
@@ -944,3 +944,16 @@
    (fn [acc x] (if (pred x) (reduced x) acc))
    nil
    coll))
+
+(defmacro case-enum
+  "Like `case`, but explicitly dispatch on Java enum ordinals."
+  [value & clauses]
+  (letfn [(enum-ordinal [e] `(let [^Enum e# ~e] (.ordinal e#)))]
+    `(case ~(enum-ordinal value)
+       ~@(concat
+          (mapcat (fn [[test result]]
+                    #_ {:clj-kondo/ignore [:discouraged-var]}
+                    [(eval (enum-ordinal test)) result])
```

Comment:
Damn this is a pity. Wonder whether we can't use reflection on the class instead?

Reply:
We *should* probably also ensure that all the dispatch enums and the input enum have the same type.

Probably jumping the shark a bit, but it would be nice if we could effectively generate a Java switch statement - especially cool if we enforce exhaustiveness too.

Reply:
Oh yeah generating switch would be cool, but absolutely no idea how to get there. As for dropping eval - if you have any ideas, I'm all ears. :)

Reply:
added some checks/tests